File: app/src/main/java/trunk/doi/base/ui/fragment/main/di/MainComponent.java
package trunk.doi.base.ui.fragment.main.di;

import dagger.BindsInstance;
import dagger.Component;

import com.base.lib.di.component.AppComponent;
import com.base.lib.di.scope.FragmentScope;

import trunk.doi.base.ui.fragment.main.di.MainModule;
import trunk.doi.base.ui.fragment.main.mvp.MainContract;
import trunk.doi.base.ui.fragment.main.MainFragment;


/**
 * ================================================
 * Description:
 * <p>
 * Created by GasMvpFragment on 11/30/2019 15:27
 * ================================================
 */

@FragmentScope
@Component(modules = MainModule.class, dependencies = AppComponent.class)
public interface MainComponent {

    void inject(MainFragment fragment);


    @Component.Builder
    interface Builder {

        @BindsInstance
        MainComponent.Builder view(MainContract.View view);

        MainComponent.Builder appComponent(AppComponent appComponent);

        MainComponent build();

    }
}

File: app/src/main/java/com/gas/app/ui/activity/main/di/MainComponent.kt
package com.gas.app.ui.activity.main.di

import dagger.Component
import com.base.lib.di.component.AppComponent
import com.base.lib.di.scope.ActivityScope

import com.gas.app.ui.activity.main.di.MainModule

import com.gas.app.ui.activity.main.MainActivity

import com.gas.app.ui.activity.main.mvp.MainContract


/**
 * ================================================
 * Description:
 * <p>
 * Created by GasMvpTemplate on 07/22/2020 20:05
 * ================================================
 */

@ActivityScope
@Component(modules = [MainModule::class], dependencies = [AppComponent::class])
interface MainComponent {
    fun inject(activity: MainActivity)
}
